### PR TITLE
Guide on how to properly extend the ShopApi

### PR DESCRIPTION
**Pull request by mamazu**
As the question has popped up how to extend the ShopApi this is a short guide on possible vectors on how to extend it.

### CODE REVIEW

**Comment by lchrusciel** [comment]
It is better than nothing, but I'm not sure about merging it as issues like #354 + #382 and so on will change a lot. I will keep it open so that people will have a reference to it.

**Comment by mamazu** [comment]
Okay, I know that the Api is currently under construction but surely we can finalize it when the concepts are more solid.

**Comment by shql** [comment]
Thanks for hinting me in the right direction here (was looking on how to adapt the shop api to our needs). As far as I can see there's a small problem with the decorated service: You are getting the cart view from the base factory and are updating this with the new property from the cart parameter. :-)

**Comment by mamazu** [comment]
What issue are you talking about exactly? I have some typos in the documentation with the parameter.

**Comment by shql** [comment]
Thanks! :-)

File: doc/ShopApiExtensions.md [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+
+        return $cart;
+    }
+}
+```
+
+If you have customized the entity that should be used in the factory, for example in this case the `OrderItem` then you can do something like the following:
+
+```php
+    public function create(OrderItem $cart, string $locale): CartSummaryView
```
Comment (GSadee) [inline doc/ShopApiExtensions.md]: `$cart` -> `$item` or `$orderItem`

**Pull request closed without merging**